Packages/com.scene.handle.extensions/Handle Extensions/Editor/Snapping/SceneHandles.Snapping.cs

```diff
@@ -533,18 +537,41 @@ public void CalculateExtents(Vector3[] localPoints)
         
         bool GetIntersectionOnAlternativePlane(Ray worldRay, Vector3 normal, Vector3 origin, out Vector3 worldPlanePosition)
         {
-            var rotation    = Quaternion.FromToRotation(normal, worldSlidePlane.normal);
             var alternativePlane = new Plane(normal, origin);
             var dist = 0.0f;
-            if (!alternativePlane.SignedRaycast(worldRay, out dist)) { worldPlanePosition = worldSlideOrigin; return false; }
+            if (!alternativePlane.SignedRaycast(worldRay, out dist))
+            {
+                worldPlanePosition = worldSlideOrigin;
+                return false;
+            }
 
-            worldPlanePosition = (rotation * (worldRay.GetPoint(dist) - origin)) + origin;
+            var tangent         = Vector3.Cross(worldSlidePlane.normal, normal);
+            if (tangent.sqrMagnitude < 0.0001f)
+            {
+                worldPlanePosition = worldSlideOrigin;
+                return false;
+            }
+            tangent.Normalize();
+            var fromQuaternion  = Quaternion.LookRotation(tangent, normal);
+            var toQuaternion    = Quaternion.LookRotation(tangent, worldSlidePlane.normal);
+            var rotation        = fromQuaternion * Quaternion.Inverse(toQuaternion);
```

Comment:
To ensure we only rotate along one axis (the axis where the two planes intersect) we take the cross product between the two planes and make sure that when we rotate, this axis stays the same